main_test.go

```diff
@@ -49,3 +52,69 @@ func TestOptionsToSlice(t *testing.T) {
 	opt.sortKeys = false
 	assert.NotContains(t, opt.ToSlice(), "-S")
 }
+
+func TestDocumentReadFrom(t *testing.T) {
+	testMsg := "hello world"
+	testReader := strings.NewReader(testMsg)
+
+	doc := &Document{
+		options: Options{
+			command: "./testdata/cat",
+		},
+	}
+
+	readCount, err := doc.ReadFrom(testReader)
+	assert.NoError(t, err)
+	assert.Equal(t, len(testMsg), int(readCount))
+}
+
+func TestDocumentWriteTo(t *testing.T) {
+	testMsg := "hello world"
+	testReader := strings.NewReader(testMsg)
+
+	doc := &Document{
+		options: Options{
+			command: "./testdata/cat",
+		},
```

Comment:
We can use `"-"` as the filter and simply `"cat"` as the command to avoid having to have a separate `cat` script:

```suggestion
		filter: "-",
		options: Options{
			command: "cat",
		},
```

No flags will be passed to `cat` since there are no options specified, so it will just run `cat -` which just spits back the input, which is exactly what we want in this case.